### PR TITLE
Call cluster/log-dump.sh from CoreDump.

### PR DESCRIPTION
**Pull request by spxtr**
Going to lean on PR Jenkins a little bit for this.

### CODE REVIEW

**Comment by spxtr** [comment]
Spiffy, looks like it worked.

@ixdy this is ready for review.

File: test/e2e/core.go [hunk 1]
```diff
@@ -17,140 +17,17 @@ limitations under the License.
...
+	"os/exec"
+	"path"
 )
 
-type command struct {
-	cmd       string
-	component string
-}
-
 func CoreDump(dir string) {
```
Comment (ixdy) [inline test/e2e/core.go]: a) can you add a docstring? I know it wasn't here before, but may as well add one now.
b) why don't we just move this to util.go? no need to have this in a separate file.

**Comment by spxtr** [comment]
Had a typo, force-push-fixed it, next run should be the proper one.

**Comment by ixdy** [comment]
LGTM

**Comment by spxtr** [comment]
@k8s bot e2e test this issue: #21940

**Comment by wojtek-t** [comment]
@spxtr - did you check that it works also in Kubemark? @gmarek - FYI

**Comment by spxtr** [comment]
@wojtek-t its behavior should be pretty similar to what was happening before. We still only attempt to ssh to the master on kubemark.

**Comment by spxtr** [comment]
@k8s bot e2e test this issue: #21940

**Comment by wojtek-t** [comment]
@spxtr - please rebase

**Comment by spxtr** [comment]
Done.

**Comment by wojtek-t** [comment]
@spxtr @ixdy - my feeling is that this really slowed down dumping logs.
As an example take a look at this run of 1000-node cluster:
http://kubekins.dls.corp.google.com/view/Scalability/job/kubernetes-e2e-gce-enormous-cluster/86/consoleText

The test itself finished around 10:02:00, and then dumping logs took more than 6 hours, and the whole run finally timed out. Can you please take a look?
@gmarek - FYI

**Comment by spxtr** [comment]
Ah, that might be right. It's pretty simple to parallelize this. I'll take care of that today.

**Comment by spxtr** [comment]
Oh, also #22670 should improve matters.

**Comment by wojtek-t** [comment]
@spxtr @ixdy - any progress on that? This is causing us a lot of pain with 1000-node clusters (those builds simply time-out, because logs collection doesn't finish in 6 hours).
If not - can we revert this one?

**Comment by spxtr** [comment]
Sorry, I got sidetracked yesterday. I'll take care of this today.

**Comment by spxtr** [comment]
@wojtek-t interestingly enough, this PR actually fixed a major bug related to the 1000-node clusters that was causing Jenkins to fall over. [This code](https://github.com/kubernetes/kubernetes/blob/b17412a1d934c85dbd1cfef9944d39bab6c75c5b/test/e2e/core.go#L105-L127) was causing it to try to run many thousands of SSHs at once, causing Jenkins to run out of file descriptors.

**Pull request merged**